packages/xod-project/src/utils.js

```diff
@@ -76,6 +79,36 @@ export const defaultValueOfType = def(
   R.flip(R.prop)(CONST.DEFAULT_VALUE_OF_TYPE)
 );
 
+export const getTypeFromLiteral = def(
+  'getTypeFromLiteral :: DataValue -> Either Error DataType',
+  literal => {
+    if (!R.is(String, literal))
+      return fail('BAD_LITERAL_VALUE', { value: literal });
+
+    if (isAmong(['True', 'False'], literal))
+      return Either.of(CONST.PIN_TYPE.BOOLEAN);
+
+    if (isAmong(R.values(CONST.INPUT_PULSE_PIN_BINDING_OPTIONS), literal))
+      return Either.of(CONST.PIN_TYPE.PULSE);
+
+    if (R.test(/^".*"$/gi, literal)) return Either.of(CONST.PIN_TYPE.STRING);
+
+    if (!isNaN(parseFloat(literal))) return Either.of(CONST.PIN_TYPE.NUMBER);
+
+    return fail('BAD_LITERAL_VALUE', { value: literal });
+  }
+);
+
+export const getTypeFromLiteralUnsafe = def(
+  'getTypeFromLiteralUnsafe :: DataValue -> DataType',
+  R.pipe(getTypeFromLiteral, explodeEither)
```

Comment:
Can’t understand how can it work?  `explodeEither` expects a potential error message, but it is not passed anywhere. Am I miss something?

Reply:
Nope, it does not:
https://github.com/xodio/xod/blob/d074a15245503c2926975250deaf6843728018d8/packages/xod-func-tools/src/monads.js#L116-L121

`explodeMaybe` does.

Reply:
Oh, I forgot an `Either` contains error out of the box. OK, don’t bother.